src/SFA.DAS.LevyTransferMatching.Infrastructure/Helpers/QuerystringHelper.cs

```diff
@@ -0,0 +1,41 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.Specialized;
+using System.Linq;
+using System.Text;
+
+namespace SFA.DAS.LevyTransferMatching.Infrastructure.Helpers
+{
+    public static class QuerystringHelper
```

Comment:
The code in this class has no unit tests.

In any case I think we could approach the problem with the following extension methods instead:

```
    public static class EnumerableStringExtensions
    {
        public static NameValueCollection ToNameValueCollection(this IEnumerable<string> value, string key)
        {
            var result = new NameValueCollection();
            foreach (var sector in value)
            {
                result.Add(key, sector);
            }

            return result;
        }
    }
```

and 

```
    public static class NameValueCollectionExtensions
    {
        public static string ToQueryString(this NameValueCollection source)
        {
            if (source.Count == 0) return string.Empty;

            var list = new List<string>();
            foreach (var key in source.AllKeys)
            {
                foreach (var value in source.GetValues(key)!)
                {
                    list.Add($"{HttpUtility.UrlEncode(key)}={HttpUtility.UrlEncode(value)}");
                }
            }

            return "?" + string.Join("&", list);
        }
    }
```

Then in the OpportunitiesService:

```
            var filters = sectors!= null ? sectors.ToNameValueCollection("sectors") : new NameValueCollection();

            var response = await _client.GetAsync($"opportunities{filters.ToQueryString()}");
```

In this way, other filters could easily be added to the nvc if present.

Reply:
these methods look good and have been replaced with current implementation